lib/kino/smart_cell/chart_builder.ex

```diff
@@ -0,0 +1,205 @@
+defmodule Kino.SmartCell.ChartBuilder do
+  use Kino.JS, assets_path: "lib/assets/chart_builder"
+  use Kino.JS.Live
+  use Kino.SmartCell, name: "Chart builder"
+
+  @as_int ["width", "height"]
+  @as_atom ["data_variable", "chart_type", "x_field_type", "y_field_type", "color_field_type"]
+
+  @impl true
+  def init(attrs, ctx) do
+    fields = %{
+      "chart_type" => attrs["chart_type"] || "bar",
+      "width" => attrs["width"],
+      "height" => attrs["height"],
+      "x_field" => attrs["x_field"],
+      "y_field" => attrs["y_field"],
+      "x_field_type" => attrs["x_field_type"],
+      "y_field_type" => attrs["y_field_type"],
+      "color_field" => attrs["color_field"],
+      "color_field_type" => attrs["color_field_type"],
+      "data_variable" => attrs["data_variable"]
+    }
+
+    ctx =
+      assign(ctx,
+        fields: fields,
+        options: %{},
+        vl_alias: nil,
+        missing_dep: missing_dep(),
+        fresh: attrs == %{}
+      )
+
+    {:ok, ctx, reevaluate_on_change: true}
+  end
+
+  @impl true
+  def scan_binding(pid, binding, env) do
+    data_options =
+      for {key, val} <- binding, is_atom(key), is_map(val), into: %{}, do: {key, Map.keys(val)}
+
+    vl_alias = vl_alias(env)
```

Comment:
When we generate the code with alias and then open the notebook, as soon as the cell starts (the first cell gets evaluated) we would generate a new code without the alias, and then update once the alias is created. I'd say it's fine, @josevalim wdyt?

Reply:
@jonatanklosko fine by me.